velox/duckdb/conversion/DuckWrapper.cpp

```diff
@@ -151,7 +154,23 @@ VectorPtr convert(
       for (auto i = 0; i < size; i++) {
         maxIndex = std::max(maxIndex, (vector_size_t)selection.get_index(i));
       }
-      auto base = convert<OP>(child, veloxType, maxIndex + 1, pool);
+      VectorPtr base;
+      // Unused dictionary elements can be uninitialized. That can cause
+      // errors if we try to decode them. Here we create a bitmap of
+      // used values to avoid that.
+      if (child.GetType() == LogicalTypeId::HUGEINT ||
```

Comment:
Nit: probably add a function here like `canHaveUnusedDictionaryElements(child.GetType())` if the bug is limited to these types.

Reply:
Uninitialized values can exist in all dictionaries, but for most types, we just copy this uninitialized memory as-is.